File: 김태일/211126/징검다리건너기.py
# 이진탐색을 사용해야 시간초과 문제 해결 가능
def solution(stones, k):
    answer = 0
    left, right = 1, max(stones)
    while left <= right:
        mid = (left + right) // 2
        jump = 0
        flag = True
        print(mid)
        for stone in stones:
            if stone - mid <= 0:
                jump += 1
                if jump == k:
                    flag = False
                    break
            else:
                jump = 0
        
        if flag:
            left = mid + 1
            answer = max(answer, mid + 1)
        else:
            right = mid - 1
    
    return answer

File: 유소연/20211126/징검다리 건너기.js
// 1, 3 테스트케이스 실패
// 효율성 실패

function solution(stones, k) {
  let count = -1;
  let maxDist = 1;
  while (maxDist <= k) {
      count++;
      let dist = 1;
      for (let i = 0; i < stones.length; i++) {
          if (stones[i] !== 0) {
              maxDist = Math.max(maxDist, dist);
              dist = 1;
              stones[i]--;
              continue;
          }
          dist++;
      }
  }
  return count;
}